tests/test/adremover-loggedout.test.js

```diff
@@ -0,0 +1,87 @@
+
+describe('The Unlock Ad Remover Paywall (logged out user)', () => {
+  beforeAll(async () => {
+    // We first need to deploy the locks
+    const addresses = []
+    // this cannot be done in parallel, or the same lock creation form gets populated with all of them at once
+    addresses.push(
+      await dashboard.deployLock(
+        locks[0].name,
+        locks[0].expirationDuration,
+        '1000',
+        locks[0].keyPrice
+      )
+    )
+
+    // save the lock address to pass it to the ad remover paywall
+    addresses.forEach((address, i) => (locks[i].address = address))
+
+    //debug.debugPage(page, true)
+    await page.goto(
+      url.paywall(
+        `/static/adremover/integrationtesting-loggedout.html?locks=${encodeURIComponent(
+          JSON.stringify(locks)
+        )}&paywall=${encodeURIComponent(url.paywall(''))}`
+      ),
+      { waitUntil: 'networkidle2' }
+    )
+  })
+
+  it('should load the data iframe and checkout iframe', async () => {
+    // 6 assertions per lock created
+    expect.assertions(6) // Assertions inside dashboard.deployLock block!
+    await wait.forIframe(2) // wait for 2 iframes to be loaded, the data and checkout iframes
+  })
+
+  it('should open the checkout UI when clicking on the button', async () => {
+    expect.assertions(1)
+    await expect(page).toClick('button', {
+      text: 'Unlock the ads free experience!',
+    })
+    // "show" is the classname that shows the checkout UI
+    await page.$('iframe[class="unlock start show"]')
+  })
+
+  it('should show the logo on the checkout UI', async () => {
+    expect.assertions(0)
+    await wait.forIframe(2) // wait for 2 iframes to be loaded, the data and checkout iframes
+    const checkoutIframe = page.mainFrame().childFrames()[1]
+    await checkoutIframe.waitForFunction(
+      unlockIcon => {
+        return !!document.body.querySelector(`img[src="${unlockIcon}"]`)
+      },
+      {},
+      unlockIcon
+    )
+  })
+
+  it('should show the "no wallet" page', async () => {
+    expect.assertions(0)
+    await wait.forIframe(2) // wait for 2 iframes to be loaded, the data and checkout iframes
+    const checkoutIframe = page.mainFrame().childFrames()[1]
+    await checkoutIframe.waitForFunction(() => {
+      const text = document.body.querySelector('p').innerText
+      return (
+        text ===
+        "To enjoy content without any ads you'll need to use a crypto-enabled browser that has a wallet. Here are a few options"
```

Comment:
I worry that this may be fragile? I guess we'll have to change this once we let the implementer use a custom message.

Reply:
this will also go away the second we have user accounts, no? However, this message is not customizable atm, as it is only displayed when there is no wallet at all